docs/varnish/vcl/varnish5.vcl

```diff
@@ -0,0 +1,269 @@
+// Varnish VCL for:
+// - Varnish 5.1 or higher (6.0LTS recommended, and is what we mainly test against)
+//   - Varnish xkey vmod (via varnish-modules package, or via Varnish Plus)
+// - eZ Platform 2.5LTS or higher with ezplatform-http-cache (this) bundle
+//
+// Make sure to at least adjust default parameters.yml, defaults there reflect our testing needs with docker.
+
+vcl 5.1;
+import std;
+import xkey;
+
+// For customizing your backend and acl rules see parameters.vcl
+include "parameters.vcl";
+
+// Called at the beginning of a request, after the complete request has been received
+sub vcl_recv {
+
+    // Set the backend
+    set req.backend_hint = ezplatform;
+
+    // Add a Surrogate-Capability header to announce ESI support.
+    set req.http.Surrogate-Capability = "abc=ESI/1.0";
+
+    // Ensure that the Symfony Router generates URLs correctly with Varnish
+    if (req.http.X-Forwarded-Proto == "https" ) {
+        set req.http.X-Forwarded-Port = "443";
+    } else {
+        set req.http.X-Forwarded-Port = "80";
+    }
+
+    // Trigger cache purge if needed
+    call ez_purge;
+
+    // Don't cache requests other than GET and HEAD.
+    if (req.method != "GET" && req.method != "HEAD") {
```

Comment:
maybe `OPTIONS` also?

Reply:
[MSDN says no](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS), but afaik in our case we should safely be able to do that for limited time. Follow Up story? Might speed up REST :)